src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPClientSideOnTypeFormattingTypedHandler.java

```diff
@@ -50,120 +52,112 @@ public Result charTyped(char charTyped,
                             @NotNull Project project,
                             @NotNull Editor editor,
                             @NotNull PsiFile file) {
-        LSPFormattingFeature formattingFeature = getFormattingFeature(file);
-        if (formattingFeature != null) {
-            boolean rangeFormattingSupported = formattingFeature.isRangeFormattingSupported(file);
+        // Gather all of the relevant client configuration
```

Comment:
Ig yiu see the old getFormattingFeatures method it checks if project is indexing. Please add this check at first

Reply:
Done.